.github/workflows/build-image.yaml

```diff
@@ -0,0 +1,74 @@
+name: Build Image
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, labeled]
+  push:
+    branches:
+      - main
+      - release/**
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
```

Comment:
Set appropriate permissions for the jobs within the action. You can do this for all jobs, or within each specific job entry.

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token